command_node.py

```diff
@@ -0,0 +1,50 @@
+# Represents a node in a tree, this node knows its parent
+class CommandNode:
```

Comment:
this class should really inherrit from dict: `class CommanNode(dict): ...`

Reply:
What reasons do you have for suggesting this?

Reply:
because most of the functionality of commandNode is providing access to its member CommandNode.children under a diffrent interface then normal python containers.

for example
``` python
class CommandNode:
    ...
    def add_child(self, child):
        self.children[child.name.upper()] = child
    ...
>>> root = CommandNode('root')
>>> root.add_child(CommandNode('node'))
```
could have just been
```python
class CommandNode(dict):
    ...
>>> root = CommandNode('root')
>>> root['node'] = CommandNode('node')
```

new maintainers won't have to learn a new name for what is essentially the same functionality.
and a lot of methods of CommandNode wont have to be written. or only when extending the functionality of existing dict methods.

Reply:
see also https://github.com/R2D2-2019/swarm_controller/pull/5#discussion_r277606667

Reply:
I agree with Isha. Due to limited time, this pull request will be closed, and the inheritance of dict will be added as a TODO.